.github/workflows/appsec.yml

```
@@ -26,168 +26,230 @@ on:
+        uses: actions/cache/restore@v4
         with:
-          path: ~/go/pkg/mod
-          key: go-pkg-mod-${{ hashFiles('**/go.sum') }}
+          path: ${{ needs.go-mod-caching.outputs.path }}
+          key: ${{ needs.go-mod-caching.outputs.key }}
           restore-keys: go-pkg-mod-
+          enableCrossOsArchive: true
+          fail-on-cache-miss: true
 
-      - name: go test
-        run: |
-          # Install gotestsum to get the results in a junit file
-          go install gotest.tools/gotestsum@latest
-          # Run the tests with gotestsum
-          env CGO_ENABLED=${{ matrix.cgo_enabled }} ${{ matrix.appsec_enabled }} /bin/bash ./.github/workflows/apps/appsec-test-contrib-submodules.sh
-
-      - name: Upload the results to Datadog CI App
-        if: matrix.distribution != 'alpine' # datadog-ci CLI doesn't work on alpine
-        uses: ./.github/actions/dd-ci-upload
-        with:
-          dd-api-key: ${{ secrets.DD_CI_API_KEY }}
-          files: ${{ env.JUNIT_REPORT }}*.xml
-          tags: go:${{ matrix.go-version }},arch:${{ runner.arch }},os:${{ runner.os }},distribution:${{ matrix.distribution }}
-
-  linux-arm64:
-    runs-on: ubuntu-latest
-    name: linux/arm64 ${{ toJSON(matrix) }}
-    strategy:
-      matrix:
-        cgo_enabled: # test it compiles with and without the cgo
-          - 0
-          - 1
-        appsec_enabled: # test it compiles with and without appsec enabled
-          - true
-          - false
-      fail-fast: false
-    steps:
-      - uses: actions/checkout@v4
-        with:
-          ref: ${{ inputs.ref || github.ref }}
-      - name: Go modules cache
-        uses: actions/cache@v4
-        with:
-          path: ~/go/pkg/mod
-          key: go-pkg-mod-${{ hashFiles('**/go.sum') }}
-          restore-keys: go-pkg-mod-
-      - name: Set up QEMU
+      - name: Set up qemu for arm64
+        if: matrix.platform == 'linux/arm64'
         uses: docker/setup-qemu-action@v3
         with:
           platforms: arm64
-      - run: |
-          env CGO_ENABLED=${{ matrix.cgo_enabled }} DD_APPSEC_ENABLED=${{ matrix.appsec_enabled }} DD_APPSEC_WAF_TIMEOUT=$DD_APPSEC_WAF_TIMEOUT ./.github/workflows/apps/appsec-test-contrib-submodules.sh docker linux/arm64
 
-  smoke-tests:
+      - name: go test
+        env:
+          CGO_ENABLED: ${{ matrix.cgo_enabled }}
+          GOMODCACHE: ${{ github.workspace }}/${{ needs.go-mod-caching.outputs.path }}
+        run: |
+          cat <<EOF | docker run -i --platform=${{ matrix.platform }}  -v "$PWD:$PWD" -w "$PWD" -v "$GOMODCACHE:$GOMODCACHE" \
+            -eGOMODCACHE -eCGO_ENABLED -eDD_APPSEC_WAF_TIMEOUT ${{ matrix.appsec_enabled }} \
+            golang:${{ matrix.go-version }}-${{ matrix.distribution }}
+              set -ex
+              if [[ ${{ matrix.distribution }} == alpine ]]; then
+                apk add gcc musl-dev libc6-compat
+              fi
+              go env
+              go test -v $TESTS
```

```suggestion
              go test -v -shuffle=on -parrallel=2 -count=4 $TESTS
```
Probably useful and won't add to much time to the CI but can make sure cleanup and parallelism is handled correctly

---

arm64 really is what I want to cover